index.js

```diff
@@ -873,6 +879,36 @@ class KeyringController extends EventEmitter {
       );
     }
   }
+
+  async _newKeyring(type, data) {
+    const keyringBuilder = this.getKeyringBuilderForType(type);
+
+    if (!keyringBuilder) {
+      return undefined;
+    }
+
+    const keyring = keyringBuilder();
+
+    await keyring.deserialize(data);
+
+    if (keyring.init) {
+      await keyring.init();
+    }
+
+    return keyring;
+  }
+}
+
+function keyringBuilderFactory(KeyringClass, BridgeClass) {
+  const builder = () => {
+    const constructorDependencies = BridgeClass ? new BridgeClass() : undefined;
+    return new KeyringClass(constructorDependencies);
+  };
+
+  builder.type = KeyringClass.type;
+
+  return builder;
 }
 
 module.exports = KeyringController;
+module.exports.keyringBuilderFactory = keyringBuilderFactory;
```

Comment:
Any strong opinions on whether we should continue exporting KeyringController as default or should instead export `KeyringController` and `keyringBuilderFactory` individually? There are already other breaking changes, so this one wouldn't be too bad.

Reply:
Moving `KeyringController` to a named exports sounds like a good idea to me

Reply:
I've made it a named export now.